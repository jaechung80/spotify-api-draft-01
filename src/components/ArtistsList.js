import React, { useEffect } from 'react';
import { Card, Container, Row, Col } from 'react-bootstrap';

const ArtistsList = (props) => {
  useEffect(() => {
    console.log('in ArtistsList props.artists', props.artists);
  });

  return (
    <Container>
      <Row>
        {props.artists?.items?.map((artist) => {
          return (
            <Col xs='3'>
              <div className='artists'>
                <Card key={artist.id} style={{ width: '18rem' }}>
                  <Card.Img
                    variant='top'
                    src={
                      artist.images.length > 0
                        ? artist.images[0].url
                        : 'https://upload.wikimedia.org/wikipedia/commons/2/26/512pxIcon-sunset_photo_not_found.png'
                    }
                  />
                  <Card.Body>
                    <Card.Title>{artist.name}</Card.Title>
                    <Card.Text>
                      Genres: {artist.genres[0] || 'not listed'}
                    </Card.Text>
                  </Card.Body>
                </Card>
              </div>
            </Col>
          );
        })}
      </Row>
    </Container>
  );
};

export default ArtistsList;
