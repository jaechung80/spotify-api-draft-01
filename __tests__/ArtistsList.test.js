import React from 'react';
import ArtistsList from '../src/components/ArtistsList';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('ArtistsList component', () => {
  const wrapper = shallow(<ArtistsList />);

  it('renders ArtistsList component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
