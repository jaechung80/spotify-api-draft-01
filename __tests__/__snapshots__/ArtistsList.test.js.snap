// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsList component renders ArtistsList component correctly 1`] = `ShallowWrapper {}`;
